src/proj_types.jl

```diff
@@ -0,0 +1,50 @@
+## Types
+
+# Projection context.  TODO: Will this be exposed?
+#type Context
+#    rep::Ptr{Void} # Pointer to internal projCtx struct
+#end
+
+@doc """
+Cartographic projection type
+""" ->
+type Projection
+    #ctx::Context   # Projection context object
+    rep::Ptr{Void} # Pointer to internal projPJ struct
+    
+    # [geod]: a structure containing the parameters of the spheroid
+    # some of the fields in [geod] are mildly duplicative of the information
+    # available in [rep], which can be exposed only through _get_spheroid_defn
+
+    # It's needed as an argument for computing great circle distances though,
+    # and for most applications, users will only have to deal with a small number
+    # of projection objects, and so we precompute it for each Projection
+    geod::geod_geodesic
```

Comment:
Yes, none of the options seem entirely desirable.  How about the following alternative: define three low level types `Proj`, `ProjCtx` and `GeodCtx` (insert better names here) which are used in the low level functions instead of passing pointers.  You're already taking this route with the `geod_geodesic` type.  In general I feel a bit bad about passing pointers around in julia, even to the low level functions.

Having defined these, a `Projection` might become
```julia
type Projection
    ctx::ProjCtx
    proj::Proj
    geod::GeodCtx
end
```

If someone really wants control, they can use the low level interface and allocate their `Proj`, `ProjCtx`, `GeodCtx` independently.

Let's not do this in the current PR, it's already pretty huge.